presto-main/src/main/java/com/facebook/presto/sql/planner/PlanBuilder.java

```diff
@@ -27,18 +29,21 @@
 class PlanBuilder
 {
     private final TranslationMap translations;
+    private final ParameterRewriter parameterRewriter;
     private final PlanNode root;
     private final Optional<Symbol> sampleWeight;
 
-    public PlanBuilder(TranslationMap translations, PlanNode root, Optional<Symbol> sampleWeight)
+    public PlanBuilder(TranslationMap translations, PlanNode root, Optional<Symbol> sampleWeight, ParameterRewriter parameterRewriter)
```

Comment:
Instead of taking a ParameterRewriter, this should just take the parameter values. It should construct a ParameterRewriter internally, as necessary, since that's an implementation detail. 

BTW, the Analysis object is available in the TranslationMap that's passed in (for the purpose of determining coercions -- arguably, this is something that should be passed separately.